rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -2931,6 +2966,21 @@ RMW_Connext_SubscriberStatusCondition::get_status(
 {
   rmw_ret_t rc = RMW_RET_ERROR;
 
+  // If condition is attached to a waitset, lock its mutex to prevent
+  // modification of the `triggered_*` flags concurrently with WaitSet::wait().
```

Comment:
I would rather take the mutex in the `get_*` functions, though I don't mind strongly.
`mutex_internal` must be taken before though, if not you cannot ensure `waitset_mutex` (now `mutex_internal` is being taken in the `get_*` functions).

Reply:
Removed in [b35d211](https://github.com/ros2/rmw_connextdds/pull/22/commits/b35d2114ef3b072f7ba386d1406eff89756d2288)